packages/template-retail-react-app/app/hooks/use-passwordless-login.js

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {AuthHelpers, useAuthHelper} from '@salesforce/commerce-sdk-react'
+
+/**
+ * This hook provides commerce-react-sdk hooks to simplify the passwordless login flow.
+ */
+export const usePasswordlessLogin = () => {
+    const authorizePasswordless = useAuthHelper(AuthHelpers.AuthorizePasswordless)
+
+    const authorizePasswordlessLogin = async (email) => {
+        await authorizePasswordless.mutateAsync({userid: email})
+    }
+
+    const login = useAuthHelper(AuthHelpers.LoginPasswordlessUser)
+
+    const loginWithPasswordlessAccessToken = async (token) => {
+        await login.mutateAsync({pwdlessLoginToken: token})
+    }
+
+    return {authorizePasswordlessLogin, loginWithPasswordlessAccessToken}
+}
+
+export default usePasswordlessLogin
```

Comment:
This is hook a candidate to be in the `commerce-sdk-react`?

Reply:
I think this hook is unnecessary wrap taking into account we already create `LoginPasswordlessUser` in useAuthHelper. It is relatively short block of code that is not likely to likely to be repeated a lot. We are not handling any custom screen/toast here either.